### PR TITLE
Convert CaseSensitiveModulesWarning to es2015

### PR DESCRIPTION
**Pull request by tejasbubane**
**What kind of change does this PR introduce?**
refactor to es2015

**Did you add tests for your changes?**
Tests for the class already present

**If relevant, link to documentation update:**

**Summary**

Refactor the function into an es2015 class

**Does this PR introduce a breaking change?**
No

**Other information**

### CODE REVIEW

File: lib/CaseSensitiveModulesWarning.js [hunk 1]
```diff
@@ -2,34 +2,42 @@
...
+"use strict";
+
+module.exports = class CaseSensitiveModulesWarning extends Error {
+	constructor(modules) {
+		super();
+		Error.captureStackTrace(this, CaseSensitiveModulesWarning);
+		this.name = "CaseSensitiveModulesWarning";
+
+		let sortedModules = this._sort(modules),
+			modulesList = this._moduleMessages(sortedModules);
```
Comment (sokra) [inline lib/CaseSensitiveModulesWarning.js]: Please use one declaration per line.

File: lib/CaseSensitiveModulesWarning.js [hunk 2]
```diff
@@ -2,34 +2,42 @@
...
+		this.message = "There are multiple modules with names that only differ in casing.\n" +
+			"This can lead to unexpected behavior when compiling on a filesystem with other case-semantic.\n" +
+			`Use equal casing. Compare these module identifiers:\n${modulesList}`;
+		this.origin = this.module = modules[0];
```
Comment (sokra) [inline lib/CaseSensitiveModulesWarning.js]: Could you use `sortedModules[0]` here?

**Comment by tejasbubane** [comment]
@sokra Changed as per you comments.

**Comment by sokra** [comment]
Thanks

**Pull request merged**